vars/wum-pipeline.groovy

```diff
@@ -0,0 +1,101 @@
+        CHANNEL_LIST='/workspace/wum_script/channel-list.txt'
+    }
+    stages {
+        stage('Preparation') {
+            steps {
+                script {
+                    echo "PATH is: $PATH"
+                    echo pwd()
+
+                    sh """
+                          echo ${JOB_CONFIG_YAML_PATH}
+                          echo '  SCENARIOS_REPOSITORY: ${SCENARIOS_REPOSITORY}' >> ${JOB_CONFIG_YAML_PATH}
+                          echo '  SCENARIOS_LOCATION: ${SCENARIOS_LOCATION}' >> ${JOB_CONFIG_YAML_PATH}
+                          echo OS_USERNAME : ${OS_USERNAME}
+                          echo OS_PASSWORD : ${OS_PASSWORD}
+                          echo UAT_APPKEY : ${UAT_APPKEY}
+                          echo LIVE_APPKEY : ${LIVE_APPKEY}
+                         
+                          #clon the apim integration repo which contains the WUM script
+                          cd ${SCENARIOS_LOCATION}
+                          if [ ! -d "${SCENARIOS_LOCATION}/apim-test-integration" ] ; then
+                              git clone ${SCENARIOS_REPOSITORY}
+                          fi
+                          cd ${SCENARIOS_LOCATION}
+
+                          # Start running WUM script which have the whole procees of calling to WUM UAT and get the Products, Version and channel list
+                          mkdir wum_script
+                          cp -f ${SCENARIOS_LOCATION}/apim-test-integration/newshell.sh ${SCENARIOS_LOCATION}/wum_script
+                          cd ${SCENARIOS_LOCATION}/wum_script 
+                          sh get-wum-uat-products.sh
+                    
+                    """
+                }
+            }
+        }
+        stage('parallel-run') {
+            steps {
+                script {
+                    try{
+                        //Converting job name list text as a json file
+                        sh """
+                            jq -Rs '[ split("\n")[] | select(length > 0)
+                                | split(",") | {JobName: .[0]} ]' "${SCENARIOS_LOCATION}/wum_script/job-list.txt" > "${SCENARIOS_LOCATION}/wum_script/job-list.json"
+                         """
+                        def inputFile = new File("${SCENARIOS_LOCATION}/wum_script/job-list.json")
+                        def jsonSulper = new JsonSlurper()
+                        def InputJSON = jsonSulper.parseText(inputFile.text)
+                        jsonSulper = null
+
+                        def wumJobs = [:]
+
+                        //Get the job name from the job-list.json and trigger builds pararelly
+                        for (int index = 0; index < InputJSON.size(); index++) {
+                            String jobName = InputJSON[index].JobName;
+                            wumJobs["job:"+ jobName] = {build job: jobName};
+                        }
+                        InputJSON = null
+                        parallel wumJobs
```

Comment:
Shouldn't we gracefully notify if the job name does not exist yet? May be it can be a slack notification.